lib/types.ts

```diff
@@ -233,3 +234,20 @@ export interface Options {
   isDockerUser?: boolean;
   config?: any;
 }
+
+export interface AutoDetectedLayers {
+  packages: PackagesFromUserInstructions;
+  layers: LayersFromUserInstructions;
+}
+
+export interface PackagesFromUserInstructions {
```

Comment:
how come this is the first time we need to define this interface? isn't it used by the regular dockerfile analysis?

Reply:
Was about to comment here since I'm not 100% on this decision. We do already have `DockerFilePackages` and `DockerFilePackages` in https://github.com/snyk/snyk-docker-plugin/blob/bf9afd0fc80fec6b0d098b321bdbf570930fa5df/lib%2Fdockerfile%2Ftypes.ts#L13
I didn't want to re-use the same types since it felt confusing, but the right thing might be to change the dockerfile ones to use the more generic type?

Reply:
what should make it more generic? it looks like `PackagesFromUserInstructions` and `DockerFilePackages` are the same (didn't talk about `DockerFileLayers`)

Reply:
Making sure I followed, you mean that `dockerfileAnalysis` should also return `PackagesFromUserInstructions`? 
Or keep it seperate?

Reply:
if `DockerFilePackages` and `PackagesFromUserInstructions` so maybe we can just use one of them?